### PR TITLE
Release `12.0.4`

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR is a follow up of [Release `v12.0.4`](https://github.com/vitessio/vitess/releases/tag/v12.0.4). The release notes are updated along with the hardcoded version strings.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: doc/releasenotes/12_0_4_release_notes.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Release of Vitess v12.0.4
+## Changelog
```
Comment (deepthi) [inline doc/releasenotes/12_0_4_release_notes.md:2]: Can you list the bugs/errors people might encounter that are fixed by this release?
The PRs may have links to issue that will help with this.

File: doc/releasenotes/12_0_4_release_notes.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### Cluster management
+
+A race condition #9819 that happens when running `PlannedReparentShard` was fixed through #9859.
+
+### Query Serving
+
+Two major bugs on `UNION` are fixed via this release. The first one, #10257, solves an issue around the `UNION` executor,
+where the execution context was not handled properly in a concurrent environment. The second one, #9945, was detected by
+one of our `UNION` tests becoming flaky in the CI, it got solved by #9979, which focuses on improve the concurrency of `UNION`
```
Comment (deepthi) [inline doc/releasenotes/12_0_4_release_notes.md:12]:
```suggestion
one of our `UNION` tests becoming flaky in the CI, it got solved by #9979, which focuses on improving the concurrency of `UNION`
```

File: doc/releasenotes/12_0_4_release_notes.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+A panic when ordering results in descending order on a `hash` vindex is also fixed. The original issue can be found here #10019. 
+
+## Changelog
+
+### Bug fixes
+#### Cluster management
+* Fix the race between PromoteReplica and replication manager tick #9859
+#### Query Serving
+* Route explain tab plan to the proper Keyspace #10028
+* Diverse fixes to UNION planning and execution #10344
```
Comment (deepthi) [inline doc/releasenotes/12_0_4_release_notes.md:24]:
```suggestion
* Multiple fixes to UNION planning and execution #10344
```

File: doc/releasenotes/12_0_4_release_notes.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+A panic when ordering results in descending order on a `hash` vindex is also fixed. The original issue can be found here #10019. 
+
+## Changelog
+
+### Bug fixes
+#### Cluster management
+* Fix the race between PromoteReplica and replication manager tick #9859
+#### Query Serving
+* Route explain tab plan to the proper Keyspace #10028
```
Comment (deepthi) [inline doc/releasenotes/12_0_4_release_notes.md:23]: does tab here mean table? If so let's change tab -> table.

**Review by deepthi — APPROVED** [review]
Minor wording changes needed. otherwise LGTM.

**Comment by frouioui** [comment]
Note to self: update the GitHub's UI release note documents when merging this PR

**Pull request merged**